include/sensors/core/tasks/capacitive_sensor_callbacks.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include "can/core/can_writer_task.hpp"
+#include "can/core/ids.hpp"
+#include "can/core/messages.hpp"
+#include "sensors/core/callback_types.hpp"
+#include "sensors/core/fdc1004.hpp"
+
+namespace capacitance_callbacks {
+using namespace fdc1004_utils;
+using namespace can_ids;
+
+/*
+ * ReadCapacitanceCallback.
+ *
+ * Struct to take a capacitance reading. The first part of sensor
+ * data is stored in the MSB register and the second part is in the
+ * LSB register. The size of the data is 24 bits.
+ */
+template <message_writer_task::TaskClient CanClient>
+struct ReadCapacitanceCallback {
+  public:
+    ReadCapacitanceCallback(CanClient &can_client, float current_offset)
+        : can_client{can_client}, current_offset{current_offset} {}
+
+    void handle_data(const sensor_callbacks::MaxMessageBuffer &MSB_buffer,
+                     const sensor_callbacks::MaxMessageBuffer &LSB_buffer) {
+        uint32_t msb_data = 0x0;
+        uint32_t lsb_data = 0x0;
+        const auto *MSB_iter = MSB_buffer.cbegin();
```

Comment:
the implied bit/byte order here sort of worries me; can we use `uint16_t` for the data elements here, and widen when we combine the two values with `MSB_SHIFT`?